### PR TITLE
Add notes about `node` usage of `name` field

### PR DESCRIPTION
**Pull request by rotu**

### CODE REVIEW

File: research/name.md [hunk 1]
```diff
@@ -41,12 +41,20 @@ Analysis of the `name` property
   - One scope must only ever point to one registry.
   - One registry can host multiple scopes.
 
+### node
+
+- Importing a package from another package is based on the path it is installed at in `node_modules`, NOT the `name` field in `package.json`.
+- The `name` field *can* be used within a package to refer to the same package<sup>[4]</sup>.
+  - The scope is part of the package name for purposes of self-reference. In a package with `"name": "@scope/foo"`, you must `import('@scope/foo')`, not e.g. `import('foo')`.
```
Comment (ljharb) [inline research/name.md]: worth noting that self-reference only covers `exports` or `imports` field entrypoints.

Comment (rotu) [reply research/name.md]: This surprised me! I think I expected a missing `exports` property to be synonymous with:
```json
  "exports": {
    ".": "index.js",
    "./*": "./*"
  }
```

File: research/name.md [hunk 2]
```diff
@@ -41,12 +41,23 @@ Analysis of the `name` property
   - One scope must only ever point to one registry.
   - One registry can host multiple scopes.
 
+### Node.js
+
+- Node.js does not require a `name` field.
+- Importing a package from another package is based on the path it is installed at in `node_modules`, NOT the `name` field in `package.json`.
+- The `name` field *can* be used within a package to refer to the same package<sup>[4]</sup>.
```
Comment (wesleytodd) [inline research/name.md]: I think this wording is fine but could use an example. I know what it means, but it took me a second, and someone who does not know what "sel-imports by name" means will likely be confused.

**Review by wesleytodd — APPROVED** [review]

File: research/name.md (hunk 2 above)
Comment (wesleytodd) [reply research/name.md]: LGTM!

**Review by Ethan-Arrowood — APPROVED** [review]
lgtm. thank you!

**Pull request merged**